compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1168,7 +1168,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             .fields
             .iter()
             .enumerate()
-            .map(|(i, field)| (field.ident.normalize_to_macros_2_0(), (i, field)))
+            .filter_map(|(i, field)| {
+                if field.vis.is_visible_locally() {
```

Comment:
This is *very* suspicious and is unlikely to be correct in presence on macro hygiene (unless this code only affects some unimportant like diagnostic details and can use heuristics).

Reply:
yep, this check is only used to filter out invisiable fileds in error reporting.

Reply:
my bad, the previous single `is_visible_locally` didn't account the private items inside local crate.
Fixed now I think? (check the visibility of field in current pat's DefTree)